software/nosql/src/main/java/brooklyn/entity/nosql/riak/RiakNodeSshDriver.java

```diff
@@ -251,6 +251,15 @@ public void customize() {
             String saveAsRiakConf = Urls.mergePaths(getRunDir(), "riak.conf");
             DynamicTasks.queueIfPossible(SshEffectorTasks.put(saveAsRiakConf).contents(riakConfTemplate));
             commands.add(sudo("mv " + saveAsRiakConf + " " + getRiakEtcDir()));
+
+            if(!osDetails.isMac()) {
+                if(!com.google.common.base.Strings.isNullOrEmpty(entity.getConfig(RiakNode.RIAK_CONFIG_FILE))) {
+                    String configTemplate = processTemplateContents(entity.getConfig(RiakNode.RIAK_CONFIG_FILE));
+                    String saveAsConfig = Urls.mergePaths(getRunDir(), "riak_extra.conf");
+                    DynamicTasks.queueIfPossible(SshEffectorTasks.put(saveAsConfig).contents(configTemplate));
```

Comment:
Driver methods are executed in effector context so it's safe to use DynamicTasks.queue (somebody correct me if I am wrong?). At the very least you should follow `queueIfPossible` with `orSubmitAsync` to be sure the task is executed.